core/controllers/contributor_dashboard.py

```diff
@@ -361,54 +360,34 @@ def _get_reviewable_exploration_opportunity_summaries(
         Returns:
             list(ExplorationOpportunitySummary). A list of the matching
             exploration opportunity summaries.
-
-        Raises:
-            Exception. No exploration_id found for the node_id.
         """
-        # 1. Fetch the eligible topics.
-        # 2. Fetch the stories for the topics.
-        # 3. Get the reviewable translation suggestion target IDs for the user.
-        # 4. Get story exploration nodes in order, filtering for explorations
-        # that have in review translation suggestions.
-        if topic_name is None:
-            topics = topic_fetchers.get_all_topics()
+        # 1. Fetch all the exploration IDs of published chapters in published
+        #    stories under the topic(s).
+        # 2. Fetch each reviewable translation suggestion to be available to the
+        #    translation reviewer.
+        # 3. Filter exploration ids. Keep those that are the target of any
+        #    fetched reviewable translation suggestions.
+        # 4. Fetch all exploration opportunity summaries for the kept
+        #    explorations.
+        # 5. If there's a pinned lesson, move the pinned lesson's summary to
+        #    the top.
+        if topic_name is None or topic_name == 'undefined':
+            topic_exp_ids = topic_services.get_published_story_exploration_ids()
         else:
             topic = topic_fetchers.get_topic_by_name(topic_name)
             if topic is None:
                 raise self.InvalidInputException(
                     'The supplied input topic: %s is not valid' % topic_name)
-            topics = [topic]
-        topic_stories = story_fetchers.get_stories_by_ids(
-            [
-                reference.story_id
-                for topic in topics
-                for reference in topic.get_all_story_references()
-                if reference.story_is_published
-            ],
-            strict=False
-        )
-        topic_exp_ids = []
-        for story in topic_stories:
-            if story is None:
-                continue
-            for node in story.story_contents.get_ordered_nodes():
-                if node.exploration_id is None:
-                    raise Exception(
-                        'No exploration_id found for the node_id: %s'
-                        % node.id
-                    )
-                topic_exp_ids.append(node.exploration_id)
+            topic_exp_ids = topic_services.get_published_story_exploration_ids(
+                topic_id=topic.id)
+
         in_review_suggestions, _ = (
```

Comment:
Is it possible to do a projection query here to fetch only the target IDs, if that's the only info we are using?

Reply:
I suppose it's possible, since the individual target IDs are `StringProperty`, which is allowed to be indexed and thus, it can be projected. Although, how do we decide which explorations are part of published stories (and their chapters)? And how do we get from the given topic name to the target IDs?

Reply:
Actually, nevermind. Ignore my last two questions. I misinterpreted your idea as a replacement to the new field in the topic summary model when your idea was more focused on fetching only the target ids as opposed to entire suggestions.

Also, is it possible for me to implement this in a separate PR if we go through with this idea?

Reply:
Correction on my first reply (in this thread): `StringProperty` is indexed by default as recorded [here](https://cloud.google.com/appengine/docs/legacy/standard/python/ndb/entity-property-reference#types).

Also, [here](https://cloud.google.com/appengine/docs/legacy/standard/python/ndb/projectionqueries#Limitations_on_projections) is my reference to the limitations of projection queries that I used to conclude that the target IDs can be projected.

Reply:
OK, sounds good. Separate PR is fine if you like but I thought the change would be a one-liner (or close to that)? Not sure.

In any case, I'm fine either way but if you want to do it in a separate PR please file an issue and link to that issue from here, and put a TODO in the code too. Thanks!

Reply:
Done.

Reply:
Adding a reference to the filed issue so we can close out this thread properly: https://github.com/oppia/oppia/issues/19664